packages/fx-core/src/core/middleware/MigrationUtils.ts

```diff
@@ -111,3 +112,146 @@ export async function needMigrateToAadManifest(ctx: CoreHookContext): Promise<bo
     return false;
   }
 }
+
+export enum FileType {
+  STATE,
+  CONFIG,
+  USERDATA,
+}
+export const fixedNamingsV3: { [key: string]: string } = {
+  "state.solution.subscriptionId": "AZURE_SUBSCRIPTION_ID",
+  "state.solution.resourceGroupName": "AZURE_RESOURCE_GROUP_NAME",
+  "state.solution.resourceNameSuffix": "RESOURCE_SUFFIX",
+  "state.fx-resource-appstudio.tenantId": "TEAMS_APP_TENANT_ID",
+  "state.fx-resource-appstudio.teamsAppId": "TEAMS_APP_ID",
+  "state.fx-resource-aad-app-for-teams.clientId": "AAD_APP_CLIENT_ID",
+  "state.fx-resource-aad-app-for-teams.clientSecret": "SECRET_AAD_APP_CLIENT_SECRET",
+  "state.fx-resource-aad-app-for-teams.objectId": "AAD_APP_OBJECT_ID",
+  "state.fx-resource-aad-app-for-teams.oauth2PermissionScopeId":
+    "AAD_APP_ACCESS_AS_USER_PERMISSION_ID",
+  "state.fx-resource-aad-app-for-teams.tenantId": "AAD_APP_TENANT_ID",
+  "state.fx-resource-aad-app-for-teams.oauthHost": "AAD_APP_OAUTH_AUTHORITY_HOST",
+  "state.fx-resource-aad-app-for-teams.oauthAuthority": "AAD_APP_OAUTH_AUTHORITY",
+  "state.fx-resource-bot.botId": "BOT_ID",
+  "state.fx-resource-bot.botPassword": "SECRET_BOT_PASSWORD",
+};
+export const provisionOutputNamingsV3: string[] = [
+  "state.fx-resource-frontend-hosting.indexPath",
+  "state.fx-resource-frontend-hosting.domain",
+  "state.fx-resource-frontend-hosting.endpoint",
+  "state.fx-resource-frontend-hosting.storageResourceId",
+  "state.fx-resource-azure-sql.sqlResourceId",
+  "state.fx-resource-azure-sql.sqlEndpoint",
+  "state.fx-resource-azure-sql.databaseName",
+  "state.fx-resource-apim.productResourceId",
+  "state.fx-resource-apim.serviceResourceId",
+  "state.fx-resource-apim.authServerResourceId",
+  "state.fx-resource-bot.skuName",
+  "state.fx-resource-bot.siteName",
+  "state.fx-resource-bot.domain",
+  "state.fx-resource-bot.appServicePlanName",
+  "state.fx-resource-bot.resourceId",
+  "state.fx-resource-bot.siteEndpoint",
+  "state.fx-resource-identity.identityName",
+  "state.fx-resource-identity.identityResourceId",
+  "state.fx-resource-identity.identityClientId",
+  "state.fx-resource-function.sku",
+  "state.fx-resource-function.appName",
+  "state.fx-resource-function.domain",
+  "state.fx-resource-function.appServicePlanName",
+  "state.fx-resource-function.functionAppResourceId",
+  "state.fx-resource-function.functionEndpoint",
+  "state.fx-resource-key-vault.m365ClientSecretReference",
+  "state.fx-resource-key-vault.botClientSecretReference",
+];
+const secretPrefix = "SECRET_";
+const configPrefix = "CONFIG__";
+const provisionOutputPrefix = "PROVISIONOUTPUT__";
+
+function generateOutputNameRegexForPlugin(pluginId: string) {
```

Comment:
I don't know when but current bicep template in V2 changed plugin id. For example, `fx-resource-frontend-hosting` is changed to `teams-tab`, but the state file still used the old plugin id `fx-resource-frontend-hosting`. We need to support this case.

Reply:
Updated, thx!

Reply:
It is changed in component update. 
Refer to the link for the mapping
https://github.com/OfficeDev/TeamsFx/blob/569d2b01b6a77856cf63f1c32b29e7e449d57e8b/packages/fx-core/src/component/migrate.ts#L95